core/domain/user_query_services.py

```diff
@@ -38,3 +40,30 @@ def save_new_query_model(
         query_status=feconf.USER_QUERY_STATUS_PROCESSING,
         user_ids=[]).put()
     return query_id
+
+
+def send_email_to_qualified_users(
+        query_id, email_subject, email_body, email_intent, email_option,
+        num_of_users_to_send):
+    """Send email to maximum 'num_of_users_to_send' qualified users."""
+    query_model = user_models.UserQueryModel.get(query_id)
+    query_model.query_status = feconf.USER_QUERY_STATUS_ARCHIVED
+
+    if email_option == feconf.USER_QUERY_EMAIL_OPTION_ALL:
+        bulk_email_model_id = email_manager.send_user_query_email(
+            query_model.submitter_id, query_model.user_ids, email_subject,
+            email_body, email_intent)
+        query_model.sent_email_model_id = bulk_email_model_id
+        query_model.put()
+    elif email_option == feconf.USER_QUERY_EMAIL_OPTION_CUSTOM:
+        recipient_ids = query_model.user_ids[:num_of_users_to_send]
```

Comment:
Btw, what's the max number of users that we can pack into user_ids before we hit the GAE memory limit? We should add a warning about that if this can plausibly occur.